frontend/src/admin/components/sidebar/dropdown/dropdown-menu.tsx

```diff
@@ -0,0 +1,35 @@
+import React, {useState} from "react";
+import MenuIcon from "./menu-icon";
+import DropdownMenuItem from "./dropdown-menu-item";
+import "./dropdown-menu.scss";
+
+export default function DropdownMenu() {
+    const [showDropdown, setShowDropdown] = useState<boolean>(false);
+
+    const toggleDropdown = () => {
+        setShowDropdown(!showDropdown);
+    };
+
+    const dismissHandler = (event: React.FocusEvent<HTMLButtonElement>): void => {
+        if (event.currentTarget === event.target) {
+            setShowDropdown(false);
+        }
+    };
+
+    return (
+        <div className="c-dropdown-menu">
+            <button
+                className={showDropdown ?"c-dropdown-menu__button--active" : "c-dropdown-menu__button"}
+                onClick={() => toggleDropdown()}
+                onBlur={(e: React.FocusEvent<HTMLButtonElement>): void => dismissHandler(e)}
+            >
+                <MenuIcon/>
+                <div className="c-dropdown-menu-items">
+                    <DropdownMenuItem id="download" name="Download user"/>
```

Comment:
Your component is named `DropdownMenu` which make it sound like it is a generic component, but here you specify that it is a User menu, which makes it quite a specific component.

You should try to stay away from "generic"/implementation names when you are not making generic components. I.e. here you are making what seems to be a `UserMenuComponent` which has the very specific responsibility of doing whatever needs to be done for the user menu. We don't care about if the menu is implemented as a dropdown or a simple list; we just want the menu for users. (also, by naming it something that references the implementation, for example `UserDropdownMenu`, you force yourself to rename it if we decide that the menu shouldn't be a dropdown, but a sliding scrollbar)

You should only use generic names when you are actually trying to make a generic component. For example a Dropdown menu. But it's important that those generic component don't have hard references to its content. The content should instead be injected through properties.
So instead of having:
```
export DropdownMenu() {
  render() {
    <div className="someDiv">
      <DrodownMenuItem id="download" name "Download user"/>
      // etc...
    </div>
  }
}
```
You would have:
```
interface Props {
  items: React.Component[] // just an example type. By injecting components you get high flexibility over what you want to displays as dropdown items
}

export DropdownMenu(props: Props) {
  render() {
    <div className="someDiv">
      {this.props.items.map(item => item)} // Pseudo code - won't compile like this
    </div>
  }
}
```
And then your specific components can utilize your generic components:
```
export UserMenu {
  downloadUser() {...}
  uploadUser() {...}

  render() {
    <DropdownMenu items={[
      <Button onClick={() => downloadUser()}>Download User</Button>
      <Button onClick={() => uploadUser()}>Upload User</Button>
    ]} />
  }
}
```

Reply:
Just to add to what @LindvedKrvang wrote. When making "generic" react components, either the child components can be passed as properties or as children:
```html
// Pass as properties
<DropdownMenu items={[ item1, item2, item3, ...]} />

// Pass as children
<DropdownMenu>
  <Item1 />
  <item2 />
  <item3 />
   ...
</DropdownMenu>
```

Passing as props give more control to the component, since it can manipulate the data before presenting it. E.g. filtering out som elements or react to an empty list. For generic components, this is often not wanted, since it imposes business logic on generic constructs.

Passing as children relinquishes the control of what is appearing inside the component, but makes it more flexible for the user.

Reply:
✅ This is changed  - i have implemented the children props